File: 2497-maximum-matching-of-players-with-trainers/2497-maximum-matching-of-players-with-trainers.cpp
class Solution {
public:
    int matchPlayersAndTrainers(vector<int>& players, vector<int>& trainers) {
        sort(players.begin(),players.end());
        sort(trainers.begin(),trainers.end());
        int count = 0;
        int i,j;
        i=0;
        j=0;
        while(i<players.size() && j<trainers.size()){
            if(trainers[j]>=players[i]){
                count++;
                i++;
                j++;
            }
            else j++;
        }
        return count;
    }
};

File: 2497-maximum-matching-of-players-with-trainers/2497-maximum-matching-of-players-with-trainers.java
import java.util.*;
class Solution {
    public int matchPlayersAndTrainers(int[] players, int[] trainers) {
        Arrays.sort(players);
        Arrays.sort(trainers);
        int count = 0;
        int i =0 , j = 0 ;
        while(i < players.length && j < trainers.length){
            if(trainers[j]>=players[i]){
                i++;
                j++;
                count++;
            }
           else  j++;
        }
        return count;
    }
}